Core/BookmarksCoreDataStorage.swift

```diff
@@ -706,6 +706,32 @@ extension BookmarksCoreDataStorage {
         })
     }
 
+    internal func containsDomain(_ domain: String,
+                                 completion: @escaping BookmarkExistsMainThreadCompletion) {
+        viewContext.perform {
+            let fetchRequest = NSFetchRequest<BookmarkManagedObject>(entityName: Constants.bookmarkClassName)
+            fetchRequest.fetchLimit = 1
+
+            fetchRequest.predicate = NSPredicate(format: "%K CONTAINS %@",
+                    #keyPath(BookmarkManagedObject.url.host),
```

Comment:
Also, I don't think this key path will work with SQLite store, as that store does not treat URLs as "objects", but uses `NSCoreDataStringSearch` function to match strings in DB columns. This probably will work in tests, where we are using in memory store - which is confusing, so we should better move to use SQLite-backed store in tests as well.

Nonetheless, what I mean is that we don't really narrow this to host only, so trying to match "test.com" in "https://example.com/index.html?q=atest.com" will most likely return true.

Reply:
Good point @bwaresiak. I took a look for an alternative and I see @THISISDINOSAUR  already wrote the logic for checking domains [here](https://github.com/duckduckgo/iOS/blob/65995ec09fab1a14321375431d0fc225c3161a99/DuckDuckGo/BookmarksManager.swift#L185)

I've taken the heart of this logic and added to / encapsulated within `BookmarksCoreDataStorage` and replaced the CoreData call

@THISISDINOSAUR I'm happy to refactor this in a separate PR?  I didn't want to include in this one + wanted to keep this PR on point.

Just one notable for all; I wrapped this new function with `MainActor` - when `Favicons` was calling this method I was consistently getting crashes while testing.  Just mentioning as perhaps its related to the crashes that were mentioned during NativeApps call

Reply:
What kind of refactoring, is there anything besides the tests?